### ISSUE TITLE
Allowed names/prefixes for unused variables

### ISSUE THREAD

**Issue by kekmitkep** [PHPCSStandards/PHP_CodeSniffer#1083]
## Is your feature request related to a problem?
When using callbacks for example for filter calls on arrays and you only use the $value, the $key parameter might be unused.

There are also multiple other use cases (like workarounds for interface implementation or deep code analysis #490 ) where this could be an issue.

This is prevalent in `UnusedFunctionParameterSniff`.

## Describe the solution you'd like
There could be reserved names or prefixes that are ignored by the unused variable/parameter check.

Should be flagged:
```php
        $selectedCompanies = $companyOptions->values()->filter(
            fn($item, $index): bool => in_array($index + 1, $selectedIndices, true)
        );
```

Should not be flagged:
```php
        $selectedCompanies = $companyOptions->values()->filter(
            fn($_, $index): bool => in_array($index + 1, $selectedIndices, true)
        );
```

## Additional context (optional)
A similar thing has already been added for PHPMD: https://github.com/phpmd/phpmd/issues/278
https://github.com/phpmd/phpmd/pull/319

For consistency's sake I would suggest a similar or the same approach.

- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [ ] I intend to create a pull request to implement this feature.

**Comment by jrfnl** [comment]
@kekmitkep Please include information about which sniff(s) this request is for.

> There are also multiple other usescases where this could be an issue.

Again, please include information about which sniffs would be involved in those usecases.

**Comment by godofcodeeyes** [comment]
Excellent contribution to open source! The community will benefit greatly from this.

**Comment by func0der** [comment]
Need the same thing. I created a PR for it.

**Comment by rodrigoprimo** [comment]
@func0der, in your use case, which sniffs are affected by this?
